pkg/aws/ec2/api/helper.go

```diff
@@ -127,9 +127,14 @@ func (h *ec2APIHelper) CreateNetworkInterface(description *string, subnetId *str
 		SubnetId:          subnetId,
 		TagSpecifications: tagSpecifications,
 	}
+	if secondaryPrivateIPCount != 0 && ipV4PrefixCount != 0 {
+		return nil, fmt.Errorf("cannot specify both secondaryPrivateIPCount %v and ipV4PrefixCount %v", secondaryPrivateIPCount, ipV4PrefixCount)
+	}
 
 	if secondaryPrivateIPCount != 0 {
 		createInput.SecondaryPrivateIpAddressCount = aws.Int64(int64(secondaryPrivateIPCount))
+	} else if ipV4PrefixCount != 0 {
+		createInput.Ipv4PrefixCount = aws.Int64(int64(ipV4PrefixCount))
 	}
```

Comment:
It's possible both would be zero since these two arguments are not required; the only required arg is the subnet id. If both are zero, the code will continue to create a default network interface.